vcr/credential/validator.go

```diff
@@ -0,0 +1,123 @@
+/*
+ * Nuts node
+ * Copyright (C) 2021 Nuts community
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ */
+
+package credential
+
+import (
+	"errors"
+	"time"
+
+	"github.com/nuts-foundation/go-did"
+)
+
+// Validator is the interface specific VC verification.
+// Every VC will have it's own rules of verification.
+type Validator interface {
+	// Validate the given credential according to the rules of the VC type.
+	Validate(credential did.VerifiableCredential) error
+}
+
+// validate the default fields
+func validate(credential did.VerifiableCredential) error {
+
+	if !containsType(credential, DefaultCredentialType) {
+		return errors.New("validation failed: 'VerifiableCredential' is required")
+	}
+
+	if !containsContext(credential, DefaultContext) {
+		return errors.New("validation failed: default context is required")
+	}
+
+	if !containsContext(credential, NutsContext) {
+		return errors.New("validation failed: nuts context is required")
+	}
+
+	if credential.ID == nil {
+		return errors.New("validation failed: 'ID' is required")
+	}
+
+	if credential.IssuanceDate.Equal(time.Time{}) {
+		return errors.New("validation failed: 'issuanceDate' is required")
+	}
+
+	if credential.Proof == nil {
+		return errors.New("validation failed: 'proof' is required")
+	}
+
+	return nil
+}
+
+func containsType(credential did.VerifiableCredential, vcType string) bool {
```

Comment:
is this something to add to `go-did` instead?

Reply:
https://github.com/nuts-foundation/go-did/pull/25